integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsToBackendWithInboundEndpointTestCase.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.axiom.om.OMElement;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.client.JMSQueueMessageProducer;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.controller.config.JMSBrokerConfigurationProvider;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.servers.ActiveMQServer;
+
+import java.io.File;
+
+/**
+ * This testcase tests consuming message from a queue and sending to a backend works
+ */
+public class JmsToBackendWithInboundEndpointTestCase extends ESBIntegrationTest {
+
+    private static final String QUEUE_NAME = "jmsQueueToHttpWithInboundEndpointTestCase";
+    private ActiveMQServer activeMQServer = new ActiveMQServer();
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        activeMQServer.startJMSBroker();
+        super.init();
+
+        OMElement esbSynapseConfig = esbUtils.loadResource(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "jms" + File.separator +
+                        "inbound" + File.separator + "transport" + File.separator + "jmsQueueToHttpInboundSynapse.xml");
+        updateESBConfiguration(esbSynapseConfig);
+
+        //Add inbound endpoint configuration
+        OMElement inboundEpConfig = esbUtils.loadResource(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "jms" + File.separator +
+                        "inbound" + File.separator + "transport" + File.separator + "jmsQueueToHttpInboundEndpoint.xml");
+        addInboundEndpoint(inboundEpConfig);
+    }
+
+    /**
+     * Test whether consuming message from a queue and sending to a backend works (i.e. JMS -> HTTP)
+     *
+     * @throws Exception if any error occurred while running tests
+     */
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Test JMS to HTTP communication with inbound endpoint")
+    public void testJmsQueueToHttpWithInboundEndpoint() throws Exception {
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewerClient.clearLogs();
+
+        //send a message to the queue
+        sendMessage();
+
+        //check for the log
+        boolean assertValue = false;
+        long startTime = System.currentTimeMillis();
+        LogEvent[] systemLogs;
+        while (!assertValue && (System.currentTimeMillis() - startTime) < 10000) {
```

Comment:
Should we derive a standard wait time for responses ? it we could globally define a value this could be changes easily

Reply:
:+1: I will extract this to a util class